consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,649 @@
+                name='validator_001',
+                id='validator_deadbeef',
+                signup_info=SignUpInfo(
+                    poet_public_key='00112233445566778899aabbccddeeff'))
+
+        # Make utils pretend it cannot deserialize the wait certificate
+        # of the new fork head
+        mock_utils.deserialize_wait_certificate.return_value = None
+
+        # create a mock_consensus_state that returns a mock with
+        # the following settings:
+        mock_state = mock.Mock()
+        mock_state.validator_signup_was_committed_too_late.return_value = False
+        mock_state.validator_has_claimed_block_limit.return_value = False
+        mock_state.validator_is_claiming_too_early.return_value = False
+        mock_state.validator_is_claiming_too_frequently.return_value = False
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_poet_enclave_module
+        mock_poet_enclave_module = mock.Mock()
+        mock_poet_enclave_module.return_value = \
+            mock_poet_enclave_factory.get_poet_enclave_module.return_value
+
+        # create mock_cur_fork_head
+        mock_cur_fork_header = \
+            mock.Mock(
+                identifier='0123456789abcdefedcba9876543210',
+                signer_pubkey='90834587139405781349807435098745',
+                previous_block_id='2',
+                header_signature='00112233445566778899aabbccddeeff')
+
+        # create mock_new_fork_head
+        mock_new_fork_header = \
+            mock.Mock(
+                identifier='0123456789abcdefedcba9876543211',
+                signer_pubkey='90834587139405781349807435098745',
+                previous_block_id='2',
+                header_signature='00112233445566778899aabbccddeeff')
+
+        # check test
+        fork_resolver = \
+            poet_fork_resolver.PoetForkResolver(
+                block_cache=mock_block_cache,
+                state_view_factory=mock_state_view_factory,
+                data_dir=self._temp_dir,
+                config_dir=self._temp_dir,
+                validator_id='validator_deadbeef')
+
+        with self.assertRaises(TypeError) as cm:
+            fork_resolver.compare_forks(
+                cur_fork_head=mock_cur_fork_header,
+                new_fork_head=mock_new_fork_header)
+            self.assertEqual(
+                'New fork head {} is not a PoET block',
+                str(cm.exception))
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.utils')
+    def test_cur_fork_head_not_valid_block(
```

Comment:
Same goes here for the method name - what you are really testing is that the current fork head is not a PoET block.